front/app/api/group_memberships/useAddMembership.ts

```diff
@@ -0,0 +1,26 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { CLErrors } from 'typings';
+import fetcher from 'utils/cl-react-query/fetcher';
+import membershipsKeys from './keys';
+import groupsKeys from 'api/groups/keys';
+import { IGroupMemberships, MembershipAdd } from './types';
+
+const addMembership = async ({ groupId, userId }: MembershipAdd) =>
+  fetcher<IGroupMemberships>({
+    path: `/groups/${groupId}/memberships`,
+    action: 'post',
+    body: { membership: { user_id: userId } },
+  });
+
+const useAddMembership = () => {
+  const queryClient = useQueryClient();
+  return useMutation<IGroupMemberships, CLErrors, MembershipAdd>({
+    mutationFn: addMembership,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: membershipsKeys.lists() });
+      queryClient.invalidateQueries({ queryKey: groupsKeys.lists() });
```

Comment:
I think it could be a good idea to also invalidate the user here. Can imagine cases where group membership changes permissions, or some other dependency. It seems like a smart thing to do.